src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading;
+using ReactiveDomain.Bus;
+
+namespace ReactiveDomain.Messaging
+{
+    /// <summary>
+    /// A command that supports cancellation via cancelation tokens
+    /// n.b. this will only work inside a single process
```

Comment:
We would have a cancel msg sent by the bus that would then trigger a cancelation source on the far side. I just walk through some patterns on this with Yves.